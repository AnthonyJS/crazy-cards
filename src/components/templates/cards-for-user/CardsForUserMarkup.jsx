import { Card, CreditAvailable } from 'components/organisms'
import { Grid } from 'components/atoms'
import Link from 'next/link'

const CardsForUserMarkup = ({
  allCards,
  cardsChosenByUser,
  possibleCardsForUser,
  cardSelectionHandler
}) => (
  <>
    <CreditAvailable
      allCards={allCards}
      cardsChosenByUser={cardsChosenByUser}
    />
    <Grid justifyItems="center">
      {possibleCardsForUser.map(card => (
        <Card
          onClick={() => cardSelectionHandler(card.id)}
          selected={cardsChosenByUser.includes(card.id)}
          key={card.id}
          {...card}
        />
      ))}
    </Grid>
    <Link href="/">
      <a>Back</a>
    </Link>
  </>
)

export default CardsForUserMarkup
