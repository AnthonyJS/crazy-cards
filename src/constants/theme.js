import { sizes } from 'constants/deviceWidths'

const breakpoints = ['0', `${sizes.m}px`, `${sizes.l}px`, `${sizes.xl}px`]

breakpoints.s = breakpoints[0]
breakpoints.m = breakpoints[1]
breakpoints.l = breakpoints[2]
breakpoints.xl = breakpoints[3]

const colors = Object.freeze({
  black: '#000e1a',
  white: '#fff',
  blue: '#007ce0',
  navy: '#004175',
  red: '#fed'
})

const theme = {
  colors: {
    primary: colors.blue,
    secondary: colors.navy,
    tertiary: colors.red,
    dark: colors.black,
    light: colors.white
  },
  space: [0, 4, 8, 16, 32, 64, 128, 256, 512],
  breakpoints
}

export default theme
