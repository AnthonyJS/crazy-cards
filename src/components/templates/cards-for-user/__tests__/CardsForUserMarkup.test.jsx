import CardsForUserMarkup from '../CardsForUserMarkup'
import cardData from 'constants/cardData'
import { render } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'
import theme from 'constants/theme'

describe('<CardsForUserMarkup />', () => {
  test('Should render all cards with none selected', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <CardsForUserMarkup
          allCards={cardData}
          cardsChosenByUser={[]}
          possibleCardsForUser={cardData}
        />
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })

  test('Should render 2 cards with one selected', () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <CardsForUserMarkup
          allCards={cardData}
          cardsChosenByUser={[1]}
          possibleCardsForUser={cardData.slice(0, 2)}
        />
      </ThemeProvider>
    )

    expect(asFragment()).toMatchSnapshot()
  })
})
