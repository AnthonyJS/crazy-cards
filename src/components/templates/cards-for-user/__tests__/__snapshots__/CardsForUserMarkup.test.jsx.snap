// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardsForUserMarkup /> Should render 2 cards with one selected 1`] = `
<DocumentFragment>
  <div
    class="FixedLabel__FixedStyle-sc-1e7rq3a-0 jrjzhp"
  >
    <div
      class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 eNZPnH"
    >
      <span
        class="Text-twcj5n-0 fxjmmM"
        color="white"
      >
        Credit available:Â 
      </span>
      <span
        class="Text-twcj5n-0 fxjmmM"
      >
        Â£1200
      </span>
    </div>
  </div>
  <div
    class="Grid-sc-1xz7oov-0 hBflZB"
  >
    <div
      class="Card__CardStyled-pjwtoy-0 dJOtkD"
    >
      <div
        class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 gyEoQt"
      >
        <span
          class="Text-twcj5n-0 gTWVCW"
        >
          Student Life
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Apr: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          18.9%
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Balance Transfer Offer Duration: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          0 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Purchase Offer Duration:Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          6 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Credit Available: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          Â£1200
        </span>
      </div>
    </div>
    <div
      class="Card__CardStyled-pjwtoy-0 fokrGT"
    >
      <div
        class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 gyEoQt"
      >
        <span
          class="Text-twcj5n-0 gTWVCW"
        >
          Anywhere Card
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Apr: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          33.9%
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Balance Transfer Offer Duration: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          0 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Purchase Offer Duration:Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          0 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Credit Available: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          Â£300
        </span>
      </div>
    </div>
  </div>
  <a
    href="/"
  >
    Back
  </a>
</DocumentFragment>
`;

exports[`<CardsForUserMarkup /> Should render all cards with none selected 1`] = `
<DocumentFragment>
  <div
    class="FixedLabel__FixedStyle-sc-1e7rq3a-0 jrjzhp"
  >
    <div
      class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 eNZPnH"
    >
      <span
        class="Text-twcj5n-0 fxjmmM"
      >
        Click the cards ðŸ’³ to choose
      </span>
    </div>
  </div>
  <div
    class="Grid-sc-1xz7oov-0 hBflZB"
  >
    <div
      class="Card__CardStyled-pjwtoy-0 fokrGT"
    >
      <div
        class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 gyEoQt"
      >
        <span
          class="Text-twcj5n-0 gTWVCW"
        >
          Student Life
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Apr: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          18.9%
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Balance Transfer Offer Duration: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          0 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Purchase Offer Duration:Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          6 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Credit Available: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          Â£1200
        </span>
      </div>
    </div>
    <div
      class="Card__CardStyled-pjwtoy-0 fokrGT"
    >
      <div
        class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 gyEoQt"
      >
        <span
          class="Text-twcj5n-0 gTWVCW"
        >
          Anywhere Card
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Apr: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          33.9%
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Balance Transfer Offer Duration: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          0 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Purchase Offer Duration:Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          0 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Credit Available: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          Â£300
        </span>
      </div>
    </div>
    <div
      class="Card__CardStyled-pjwtoy-0 fokrGT"
    >
      <div
        class="Box-sc-141iqn8-0 Alignment-kk8fl8-0 gyEoQt"
      >
        <span
          class="Text-twcj5n-0 gTWVCW"
        >
          Liquid Card
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Apr: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          33.9%
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Balance Transfer Offer Duration: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          12 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Purchase Offer Duration:Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          6 Months
        </span>
      </div>
      <div>
        <span
          class="Text-twcj5n-0 eyBDVE"
        >
          Credit Available: Â 
        </span>
        <span
          class="Text-twcj5n-0 hEKyEh"
        >
          Â£3000
        </span>
      </div>
    </div>
  </div>
  <a
    href="/"
  >
    Back
  </a>
</DocumentFragment>
`;
